pkg/service/model_build_target_group.go

```diff
@@ -332,19 +332,25 @@ func (t *defaultModelBuildTask) buildTargetGroupHealthCheckUnhealthyThresholdCou
 	return unhealthyThresholdCount, nil
 }
 
-func (t *defaultModelBuildTask) buildTargetType(_ context.Context) (elbv2model.TargetType, error) {
+func (t *defaultModelBuildTask) buildTargetType(_ context.Context, port corev1.ServicePort) (elbv2model.TargetType, error) {
 	svcType := t.service.Spec.Type
 	var lbType string
 	_ = t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixLoadBalancerType, &lbType, t.service.Annotations)
 	var lbTargetType string
+	if t.service.Spec.LoadBalancerClass != nil {
```

Comment:
maybe just remove the check if t.service.Spec.LoadBalancerClass != nil, since the service will only be called when it's have one of the `SvcLBSuffixTargetType` annotation any way if it's with external LB type.

Reply:
ack